rust/opendp-ffi/src/util.rs

```diff
@@ -228,43 +227,54 @@ pub fn into_box<T, U>(o: T) -> Box<U> {
     unsafe { Box::from_raw(p) }
 }
 
-pub fn into_owned<T>(p: *mut T) -> T {
-    assert!(!p.is_null());
-    *unsafe { Box::<T>::from_raw(p) }
+pub fn into_owned<T>(p: *mut T) -> Option<T> {
```

Comment:
I was using an option so that a more specialized message can be attached at call sites. It should be possible to map_err and mutate, or just let the errors be less specific. I'm ok with any of these routes.

Reply:
On second thought, the additional information provided by the more specialized message is already captured in the backtrace.

Reply:
Yeah seems like the error is always basically the same (passing a null pointer), so with the backtrace it should be enough.